File: generated-code/codecraft/csharp/TcpReadWrite/Runner.cs
using System;
using System.IO;
using System.Net.Sockets;

namespace TransGenTest
{
    public class Runner
    {
        public static void Main(string[] args)
        {
            if (args.Length != 2)
            {
                throw new Exception("Pass host and port as parameters");
            }
            string host = args[0];
            int port = int.Parse(args[1]);

            using (var tcpClient = new TcpClient(host, port))
            {
                using (var stream = new BufferedStream(tcpClient.GetStream()))
                {
                    var reader = new BinaryReader(stream);
                    Model.PlayerView input = Model.PlayerView.ReadFrom(reader);
                    Console.WriteLine(input.ToString());
                    var writer = new BinaryWriter(stream);
                    input.WriteTo(writer);
                    writer.Flush();
                }
            }
        }
    }
}

File: generated-code/fsharp/codecraft/TcpReadWrite/Runner.fs
namespace TransGenTest

open System
open System.IO
open System.Net.Sockets

module Runner =

    [<EntryPoint>]
    let main argv =
        if argv.Length <> 2 then
            failwith "Pass host and port as parameters"
        let host = argv.[0]
        let port = argv.[1] |> Int32.Parse

        use client = new TcpClient(host, port)
        let stream = new BufferedStream(client.GetStream())

        let reader = new BinaryReader(stream)
        let input = Model.PlayerView.readFrom(reader)

        Console.WriteLine(input)

        let writer = new BinaryWriter(stream)
        input.writeTo(writer)
        writer.Flush()

        0